src/components/Card.js

```diff
@@ -1,32 +1,38 @@
 export default class Card {
-  constructor(data, cardSelector, handleImageClick) {
+  constructor(
+    data,
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  ) {
     this._data = {
       name: data.name,
       link: data.link,
+      id: data._id,
+      isLiked: data.isLiked,
     };
     this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick;
     this._cardSelector = cardSelector;
+    this._handleLikeClick = handleLikeClick;
   }
 
   _setEventListeners() {
-    this._likeIcon.addEventListener("click", this._handleLikeIcon);
+    this._likeIcon.addEventListener("click", () => {
+      this._handleLikeClick(this._likeIcon, this._data.isLiked, this._data.id);
+      this._data.isLiked = !this._data.isLiked;
```

Comment:
`      this._data.isLiked = !this._data.isLiked;
`

You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.